File: Heat/ModelBuilders/ManifacturerModelViewBuilder.vb
Imports Heat.ViewModels.ManifacturerModels
Imports Heat.Models
Imports System.Data.Entity

Public Class ManifacturerModelViewBuilder
    Private _db As IHeatDBContext

    Sub New(context As IHeatDBContext)
        _db = context
    End Sub

    Public Function GetIndexManifacturerModelViewModel() As List(Of IndexManifacturerModelViewModel)
        Dim result As List(Of IndexManifacturerModelViewModel)

        result = _db.ManifacturerModels.
            Include(Function(x) x.Manifacturer).
            Select(Function(x) New IndexManifacturerModelViewModel With {.ID = x.ID, .Model = x.Model, .Manifacturer = x.Manifacturer.Name}).ToList

        Return result


    End Function

    Public Function GetCreateManifacturerModelViewModel() As CreateManifacturerModelViewModel
        Dim result As New CreateManifacturerModelViewModel

        result.ManifacturerList = _db.Manifacturers.ToList.OrderBy(Function(x) x.Name).ToSelectListItems(Function(x) x.Name, Function(x) x.ID, "")

        Return result
    End Function

    Public Function GetEditManifacturerModelViewModel(id) As EditManifacturerModelViewModel
        Dim result As New EditManifacturerModelViewModel
        Dim dbItem As ManifacturerModel

        dbItem = _db.ManifacturerModels.Find(id)

        result.ID = id
        result.ManifacturerID = dbItem.ManifacturerID
        result.Model = dbItem.Model
        result.ManifacturerList = _db.Manifacturers.ToList.OrderBy(Function(x) x.Name).ToSelectListItems(Function(x) x.Name, Function(x) x.ID, result.ManifacturerID)

        Return result

    End Function
End Class

File: Heat.ConvertedToC#/ModelBuilders/ManifacturerModelViewBuilder.cs
using System.Collections.Generic;
using System.Linq;
using Heat.ViewModels.ManifacturerModels;
using Heat.Models;
using System.Data.Entity;
namespace Heat
{

    public class ManifacturerModelViewBuilder
	{

		private IHeatDBContext _db;
		public ManifacturerModelViewBuilder(IHeatDBContext context)
		{
			_db = context;
		}

		public List<IndexManifacturerModelViewModel> GetIndexManifacturerModelViewModel()
		{
			List<IndexManifacturerModelViewModel> result = null;

			result = _db.ManifacturerModels.Include(x => x.Manifacturer).Select(x => new IndexManifacturerModelViewModel {
				ID = x.ID,
				Model = x.Model,
				Manifacturer = x.Manifacturer.Name
			}).ToList();

			return result;


		}

		public CreateManifacturerModelViewModel GetCreateManifacturerModelViewModel()
		{
			CreateManifacturerModelViewModel result = new CreateManifacturerModelViewModel();

			result.ManifacturerList = _db.Manifacturers.ToList().OrderBy(x => x.Name).ToSelectListItems(x => x.Name, x => x.ID.ToString(), "");

			return result;
		}

		public EditManifacturerModelViewModel GetEditManifacturerModelViewModel(int id)
		{
			EditManifacturerModelViewModel result = new EditManifacturerModelViewModel();
			ManifacturerModel dbItem = null;

			dbItem = _db.ManifacturerModels.Find(id);

			result.ID = id;
			result.ManifacturerID = dbItem.ManifacturerID;
			result.Model = dbItem.Model;
			result.ManifacturerList = _db.Manifacturers.ToList().OrderBy(x => x.Name).ToSelectListItems(x => x.Name, x => x.ID.ToString(), result.ManifacturerID.ToString());

			return result;

		}
	}
}
